### ISSUE TITLE
[BUG] Changed case of foldernames

### ISSUE THREAD

**Issue by nalim** [SamSchott/maestral#176]
**Describe the bug**
I changed the case of names of several folders synced by Dropbox on one of my Windows 10 (64b) machines :

"Folder" -> "folder" etc.

later I started another machines with Fedora Linux 32 and Maestral 1.1.0 and Maestral data folder on an NTFS partition. Maestral didn't detect the change in case of the renamed folders but started to download a new folder (including its content) with the changed name:

"folder"  // new folder
"Folder" // "old" folder

so I ended up with duplicated folder structure.  It happened on 2 of my Linux boxes, another 5 or so didn't experience this problem.
 
**To Reproduce**
I was not able to reproduce the problem.

**Expected behaviour**
The folder with change of case in its name should be renamed, not duplicated.

**System (please complete the following information):**
 - Maestral version: 1.1.0
 - OS: Fedora 32
 - Desktop environment: Cinnamon Desktop
 - Qt version:

**Additional context**
Maestral/Dropbox data shared between Windows and Linux using an NFTS partition

**Comment by SamSchott** [comment]
Thanks for the report!

The current behaviour of Maestral should be to stick with the casing of the name when it was first downloaded. What you are seeing is definitely a bug. Could you check if the filesystem is correctly identified as case-sensitive on the affected machines? What does the following Python snippet return?

```Python
from maestral.main import Maestral

m = Maestral()
print(m.sync.is_case_sensitive)
```

Maestral does not currently sync changes in filename when only the casing is changed. This is on my todo list...

**Comment by nalim** [comment]
$ python
Python 3.8.5 (default, Aug 12 2020, 00:00:00) 
[GCC 10.2.1 20200723 (Red Hat 10.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from maestral.main import Maestral
>>> m = Maestral()
>>> print(m.sync.is_case_sensitive)
True
>>>

**Comment by SamSchott** [comment]
Hrm, so this seems to work fine. Somehow, the conversion from the relative Dropbox path (not case-sensitive) to the local path (case-sensitive) seems to fail. If you manage to reproduce the issue, could you set the logger to debug (`maestral log level DEBUG`) and post the log output for the relevant sync events (`maestral log show`)?

**Comment by nalim** [comment]
Ok, I wll try to reproduce the issue but it won't be easy. I alreadydid some tests but each time the change in case was handled properly. If you wrote "Somehow, the conversion from the relative Dropbox path (not case-sensitive) to the local path (case-sensitive) seems to fail", does it mean Dropbox don't distinguish between "Folder" and "folder"? Or what do you mean? When I changed the casing from "Folder" to "folder" while in Windows, the change was properly propagated/synced to Dropbox servers and to all my other Windows machines...

**Comment by SamSchott** [comment]
Exactly, Dropbox does not distinguish between "Folder" and "folder". Both represent the same item on Dropbox servers and any difference is only cosmetic, i.e., how the item is displayed to the user. Try creating two folders on Dropbox with the same name but different casing. It won't let you...

Despite not being case-sensitive, Dropbox remembers the casing of the file / folder name in its metadata. However, the public API provides the correct casing for the item name only and not for any parent folders. Therefore, a file at '/Folder/Notes.txt' may appear through the API as '/folder/Notes.txt'. Maestral now pieces together the correct path because it previously synced the parent folder and therefore knows its correct casing. Apparently this failed for some reason.

If you encounter any issues with this again, let me know. Otherwise, don't worry too much about reproducing this. I am in the process of overhauling the local index anyways. Part of that will be a refactored lookup of correctly cased paths.

**Comment by SamSchott** [comment]
This should be fixed in v1.2.0 with an overhaul of case handling.
